src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,454 @@
+                  ASSIGN        Length lhs_Term rhs_Term
+                  BLOCK         Length expr_Term Stat*
+                  IF            Length cond_Term then_Term else_Term
+                  CLOSURE       Length meth_Term target_Type env_Term*
+                  MATCH         Length sel_Term CaseDef*
+                  RETURN        Length meth_ASTRef expr_Term?
+                  TRY           Length expr_Term CaseDef* finalizer_Term?
+                  THROW         Length expr_Term
+                  SEQLITERAL    Length elem_Term*
+                  JSEQLITERAL   Length elem_Term*
+                  BIND          Length boundName_NameRef patType_Type pat_Term
+                  ALTERNATIVE   Length alt_Term*
+                  UNAPPLY       Length fun_Term ImplicitArg* pat_Term*
+                  ANNOTATED     Length annot_Term underlying_Term
+                  EMPTYTREE
+
+  CaseDef       = CASEDEF       Length pat_Tree guard_Tree rhs_Tree
+  ImplicitArg   = IMPLICITARG   Length arg_Tree
+  Template      = TEMPLATE      Length parent_Tree* SelfDef? Stat*
+// if there is a primary constructor, it is the first statement in Stat*..
+  SelfDef       = Param
+  ASTRef        = Nat                          		// byte position in AST payload
+
+  Path          = Constant
+                  TERMREFdirect        sym_ASTRef
+                  TERMREFstatic        fullyQualified_NameRef
+                  TERMREFsymbol        qual_Type sym_ASTRef
+                  TERMREF              qual_Type possiblySigned_NameRef
+                  THIS          Length clsRef_Type
+                  SHARED               path_ASTRef
+
+  Constant      = UNITconst
+                  FALSEconst
+                  TRUEconst
+                  BYTEconst            Nat
+                  BYTEneg              NegNat
+                  SHORTconst           Nat
+                  SHORTneg             NegNat
+                  CHARconst            Nat
+                  INTconst             Nat
+                  INTneg               NegNat
+                  LONGconst            LongNat
+                  LONGneg              NegLongNat
+                  FLOATconst           LongNat
+                  DOUBLEconst          LongNat
+                  STRINGconst          NameRef
+                  NULLconst
+                  CLASSconst    Length Type
+                  ENUMconst     Length Path
+  NegNat        = Nat          					// negValue = -natValue - 1
+  NegLongNat    = LongNat     					// negValue = -natValue - 1
+
+  Type          = Path
+                  TYPEREFdirect        sym_ASTRef
+                  TYPEREFstatic        fullyQualified_NameRef
+                  TYPEREFsymbol        qual_Type sym_ASTRef
+                  TYPEREF              qual_Type possiblySigned_NameRef
+                  SUPERtype     Length this_Type underlying_Type
+                  SKOLEMtype    Length underlying_Type
```

Comment:
Yes, as `this` references in refined types. Also, Scala2 existential types will be mapped to skolem types.